File: cs/cg_cs/src/rendering/RenderVertex.cs
using OpenTK;
using OpenTK.Graphics;

/**
 * Helping data structure to represent a render vertex in a @VertexBufferObject.
 * 
 *
 */
namespace computergraphics
{
	public class RenderVertex
	{
		/**
		 * 3D position.
		 */
		private Vector3 position;

		/**
		 * 3D normal.
		 */
		private Vector3 normal;

		/**
		 * Texture coordinate
		 * */
		private Vector2 texCoord;

		/**
		 * 4D color.
		 */
		private Color4 color;

		public Vector3 Position
		{
			get { return position; }
		}

		public Vector3 Normal
		{
			get { return normal; }
		}

		public Color4 Color
		{
			get { return color; }
		}

		public Vector2 TexCoord
		{
			get { return texCoord; }
		}

		public RenderVertex(Vector3 position, Vector3 normal, Color4 color) : this(position, normal, color, new Vector2(0, 0))
		{
			
		}

		public RenderVertex(Vector3 position, Vector3 normal, Color4 color, Vector2 texCoord)
		{
			this.position = position;
			this.normal = normal;
			this.color = color;
			this.texCoord = texCoord;
		}
	}
}

File: android/lib_shared/src/main/java/edu/hawhamburg/shared/rendering/RenderVertex.java
/**
 * Diese Datei gehört zum Android/Java Framework zur Veranstaltung "Computergrafik für
 * Augmented Reality" von Prof. Dr. Philipp Jenke an der Hochschule für Angewandte
 * Wissenschaften (HAW) Hamburg. Weder Teile der Software noch das Framework als Ganzes dürfen
 * ohne die Einwilligung von Philipp Jenke außerhalb von Forschungs- und Lehrprojekten an der HAW
 * Hamburg verwendet werden.
 *
 * This file is part of the Android/Java framework for the course "Computer graphics for augmented
 * reality" by Prof. Dr. Philipp Jenke at the University of Applied (UAS) Sciences Hamburg. Neither
 * parts of the framework nor the complete framework may be used outside of research or student
 * projects at the UAS Hamburg.
 */
package edu.hawhamburg.shared.rendering;

import edu.hawhamburg.shared.math.Vector;

/**
 * Helping data structure to represent a render vertex in a @VertexBufferObject.
 * 
 * @author Philipp Jenke
 */
public class RenderVertex {

  public RenderVertex(Vector position, Vector normal, Vector color) {
    this(position, normal, color, new Vector(0,0));
  }
  
  public RenderVertex(Vector position, Vector normal, Vector color, Vector texCoords) {
    this.position = position;
    this.normal = normal;
    this.color = color;
    this.texCoords = texCoords;
  }

  /**
   * 3D position.
   */
  public Vector position;

  /**
   * 3D normal.
   */
  public Vector normal;

  /**
   * 4D color.
   */
  public Vector color;
  
  /**
   * 2D Texture coordinate
   */
  public Vector texCoords;

}
